emscripten.py

```diff
@@ -484,7 +484,7 @@ def create_module_asmjs(function_table_sigs, metadata,
   asm_runtime_thread_local_vars = create_asm_runtime_thread_local_vars()
 
   stack = ''
-  if not (shared.Settings.WASM and shared.Settings.SIDE_MODULE):
+  if not shared.Settings.RELOCATABLE and not (shared.Settings.WASM and shared.Settings.SIDE_MODULE):
```

Comment:
IIUC this is redundant. 

If we apply De Morgan's here I a little clearer to see:

```
if shared.Settings.RELOCATABLE or (shared.Settings.WASM and shared.Settings.SIDE_MODULE):
```

Since RELOCATABLE is always set for SIDE_MODULE the second clause is redundant I think?

Reply:
In theory, we may want to allow non-relocatable dynamic modules (i.e., loadable at runtime, but at known-beforehand locations in memory/table). At least it would be nice to not rule that out for now, I think. So checking for relocation separately makes sense in my opinion.

Reply:
Yeah, that's how my original working diff was. I changed it to mirror the logic in the other part of the diff when I realized that other change was necessary.